src/ssmDocument/explorer/documentItemNodeWriteable.ts

```diff
@@ -0,0 +1,39 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { SSM } from 'aws-sdk'
+import { RegistryItemNode } from './registryItemNode'
+import { SsmDocumentClient } from '../../shared/clients/ssmDocumentClient'
+import { DocumentItemNode } from './documentItemNode'
+
+export class DocumentItemNodeWriteable extends DocumentItemNode {
+    public constructor(
+        documentItem: SSM.Types.DocumentIdentifier,
+        public readonly client: SsmDocumentClient,
+        public readonly regionCode: string,
+        public readonly parent: RegistryItemNode
+    ) {
+        super(documentItem, client, regionCode)
+        this.contextValue = 'awsDocumentItemNodeWriteable'
+        this.parent = parent
+    }
+
+    public async deleteDocument(): Promise<SSM.Types.DeleteDocumentResult> {
+        if (!this.documentName || !this.documentName.length) {
+            return Promise.resolve({})
```

Comment:
Should this be considered an error state? Doesn't look like this is handled for telemetry, at the very least. Could also return `undefined`. (same comment for `updateDocumentVersion` below)

Reply:
Will handle this as error state